mirdata/rwc_genre.py

```diff
@@ -0,0 +1,237 @@
+RWC_GENRE_DIR = 'RWC-Genre'
+RWC_GENRE_ANNOT_REMOTE_1 = utils.RemoteFileMetadata(
+    filename='AIST.RWC-MDB-G-2001.BEAT.zip',
+    url='https://staff.aist.go.jp/m.goto/RWC-MDB/AIST-Annotation/AIST.RWC-MDB-G-2001.BEAT.zip',
+    checksum='66427ce5f4485088c6d9bc5f7394f65f')
+RWC_GENRE_ANNOT_REMOTE_2 =  utils.RemoteFileMetadata(
+    filename='AIST.RWC-MDB-G-2001.CHORUS.zip',
+    url='https://staff.aist.go.jp/m.goto/RWC-MDB/AIST-Annotation/AIST.RWC-MDB-G-2001.CHORUS.zip',
+    checksum='e9fe612a0ddc7a83f3c1d17fb5fec32a')
+
+
+RWCGenreTrack = namedtuple(
+    'RWCGenreTrack',
+    ['track_id',
+     'audio_path',
+     'sections',
+     'beats',
+     'duration_sec',
+     'title',
+     'artist',
+     'variation',
+     'instruments']
+)
+
+
+def download(data_home=None, clobber=False):
+    save_path = utils.get_save_path(data_home)
+    dataset_path = os.path.join(save_path, RWC_GENRE_DIR, 'annotations')
+
+    if clobber:
+        utils.clobber_all(RWC_GENRE_ANNOT_REMOTE_1,
+                          dataset_path,
+                          data_home)
+        utils.clobber_all(RWC_GENRE_ANNOT_REMOTE_2,
+                          dataset_path,
+                          data_home)
+
+    if utils.check_validated(dataset_path):
+        print("""
+                The {} dataset has already been downloaded and validated.
+                Skipping download of dataset. If you feel this is a mistake please
+                rerun and set clobber to true
+                """.format(RWC_GENRE_DIR))
+        return
+
+    download_path_1 = utils.download_from_remote(
+        RWC_GENRE_ANNOT_REMOTE_1, data_home=data_home, clobber=clobber)
+    download_path_2 = utils.download_from_remote(
+        RWC_GENRE_ANNOT_REMOTE_2, data_home=data_home, clobber=clobber)
+
+    if not os.path.exists(dataset_path):
+        os.makedirs(dataset_path)
+
+    utils.unzip(download_path_1, dataset_path, cleanup=True)
+    utils.unzip(download_path_2, dataset_path, cleanup=True)
+
+    missing_files, invalid_checksums = validate(dataset_path, data_home)
+    if missing_files or invalid_checksums:
+        print("""
+            Unfortunately the audio files of the RWC-Genre dataset are not available
+            for download. If you have the RWC-Genre dataset, place the contents into a
+            folder called RWC-Genre with the following structure:
+                > RWC-Genre/
+                    > annotations/
+                    > audio/
+            and copy the RWC-Genre folder to {}
+        """.format(save_path))
+
+    # metadata
+    # utils.download_from_remote(
+    #         RWC_GENRE_METADATA, data_home=data_home, clobber=clobber)
+
+
+def validate(dataset_path, data_home=None):
+    missing_files, invalid_checksums = utils.validator(RWC_GENRE_INDEX, data_home, dataset_path)
+    return missing_files, invalid_checksums
+
+
+def track_ids():
+    return list(RWC_GENRE_INDEX.keys())
+
+
+def load(data_home=None):
+    validate(data_home)
```

Comment:
Same comments as above about this not working because validate takes two arguments.